proxy/frontend_handler.go

```diff
@@ -55,6 +55,7 @@ func (h *FrontendHandler) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 
 	// Send response messages to client
 	go func() {
+		defer closeConnection(ws)
```

Comment:
This is the actual fix to the test deadlock. I defer this same method above on line 46. But it needs to be called in this situation too, where the backend closes first.